http.go

```diff
@@ -92,7 +103,7 @@ func (client *Client) Request(rawPayload []byte, method, route string, query url
 				return nil, nil, err
 			}
 		}
-		resp, err := (&http.Client{}).Do(callSummary.HTTPRequest)
+		resp, err := DefaultHTTPClient.Do(callSummary.HTTPRequest)
```

Comment:
@walac I think it doesn't really matter either way. If we later call more methods, we can easily switch to an interface, and likewise, we could also add extra function types to the client. But if you anticipate already that we'll be calling more methods of the client, I don't object to going with an interface.